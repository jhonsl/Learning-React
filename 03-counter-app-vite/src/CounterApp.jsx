import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value );

  const incrementCounter = () => setCounter( (value) => value + 1 );
  // const incrementCounter = () => setCounter( counter + 1 );
  const decrementCounter = () => setCounter( (value) => value - 1 );
  // const decrementCounter = () => setCounter( counter - 1 );
  const resetCounter = () => setCounter( value );

  return (
    <>
      <h1>Counter App</h1>
      <h2 data-testid="title">El valor es: { counter }</h2>
      <button aria-label='btn-add' onClick={ incrementCounter }>
        +1
      </button>
      <button aria-label='btn-rest' onClick={ decrementCounter }>
        -1
      </button>
      <button aria-label='btn-reset' onClick={ resetCounter }>
        Reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}