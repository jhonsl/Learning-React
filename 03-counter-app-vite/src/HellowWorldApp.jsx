import PropTypes from 'prop-types';

const message = {
    title: 'hola mundo',
    subTitle: 'mi nombre es jhon'
};
const numeros = [1,2,3,4,5,6,7,8,9];
const sumar = (numero1, numero2) => numero1 + numero2;

export const HellowWorldApp = ({ title, subTitle }) => {
    return (
        <>
            <h1>{ message.title }</h1>
            <code>{ JSON.stringify(message) }</code>
            <p>{ message.subTitle }</p>
            <p>{ numeros }</p>

            <hr />

            <h2>El resultado de la suma es: { sumar(4,5) }</h2>
            <h2>{ title }</h2>
            <h3>{ subTitle }</h3>
        </>
    )
}

HellowWorldApp.defaultProps = {
    title: 'mil a√±os de soledad',
    subTitle: 2002
}

HellowWorldApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number.isRequired
}