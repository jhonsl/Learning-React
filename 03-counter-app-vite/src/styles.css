html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 10px 15px 10px 15px;
    margin-right: 20px;
    font-size: 1.3rem;
}