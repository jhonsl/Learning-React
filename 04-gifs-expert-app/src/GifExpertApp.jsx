import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Dragon Ball']);
  const onAddCategory = ( category ) => {

    if( !categories.includes( category ) ) {
      setCategories([...categories, category])
    }
  };
 
  return (
    <>
      <h1>GifsExpertApp</h1> 

      <AddCategory onNewCategory={ onAddCategory } />

      { 
        categories.map( category => (
          <GifGrid 
            key={ category } 
            category={ category }
          />
        )) 
      }
    </>
  )
}

export default GifExpertApp;