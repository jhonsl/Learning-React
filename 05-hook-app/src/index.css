body {
  padding: 20px;
}

button {
  margin-right: 10px;
}

/* a {
  border-right: 2px solid black;
  margin-left: 1%;
  padding-right: 1%;
} */