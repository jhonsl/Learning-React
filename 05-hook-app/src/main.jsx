import React from 'react'
import ReactDOM from 'react-dom/client'
import { Padre } from './07-tarea-memo/Padre'

import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Padre />
  </React.StrictMode>
)
