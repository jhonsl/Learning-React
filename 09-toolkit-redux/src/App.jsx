import { useSelector, useDispatch } from 'react-redux';
import { decrement, increment, incrementBy } from './store/slices/counter/counterSlice';

import './App.css'

function App() {
  const { counter } = useSelector(state => state.counter);
  const dispatch = useDispatch();

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src="/vite.svg" className="logo" alt="Vite logo" />
        </a>
      </div>
      <div className="card">
        <button>
          count is { counter }
        </button>
      </div>
      <div>
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>

        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>

        <button onClick={() => dispatch(incrementBy(22))}>
          Increment by 2
        </button>
      </div>
    </div>
  )
}

export default App
