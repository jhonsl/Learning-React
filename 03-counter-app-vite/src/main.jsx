import React from 'react';
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp';
import './styles.css'

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={10}/>
        {/* <FirstApp title='Esto es un titulo'/> */}
    </React.StrictMode>
)  