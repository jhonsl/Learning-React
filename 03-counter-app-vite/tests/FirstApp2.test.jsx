import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const title = 'Hola soy goku';
    const subTitle = "Hola soy un subtitulitooooo";
    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <FirstApp title={ title }/> );
        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el titulo Hola soy goku', () => {
        render( <FirstApp title={ title }/> );
        expect( screen.getByText( title )).toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render( <FirstApp title={ title }/> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML  ).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render( 
            <FirstApp 
                title={ title }
                subTitle={subTitle}
            /> 
        );
        expect( screen.getAllByText( subTitle ).length  ).toBe( 2 );
    });
});