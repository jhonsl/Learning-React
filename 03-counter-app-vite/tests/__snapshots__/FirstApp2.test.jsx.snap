// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <p>
    Esto es un subtitulo
  </p>
  <p>
    Esto es un subtitulo
  </p>
  <p>
    Jhon Salazar
  </p>
</div>
`;
